documentation/smg/16-serialiser.md

```diff
@@ -0,0 +1,275 @@
+# Serialisation details
+
+As of the integration of the C++ serialiser, the serialised format has been updated to be more consistent and to have greater capabilities.
+
+For reference, the old format is also detailed below
+
+## New format
+
+### Standard format
+The new style has a standardised header format depending on the shape of the data (with the exception of function handles).
+This format allows extension to support arrays of objects and structs in a way which the old serialiser couldn't.
```

Comment:
I see "header" regarding data transfer, as only possibly being a preceding chunk which is not the data which outlines how the data should be interpreted. Am I not outlining the structure below?

Reply:
What I'm saying is that you include "Data" in your description of the header. I'm expecting that Data will follow the header but not be part of it. I'm suggesting that you remove Data from the header description, and, if it is needed, say something like "the header will be followed by the data it describes; these headers include all information on the data length so that no "footer" information is required to tag end of data."